charts/zookeeper/templates/zookeeper.yaml

```diff
@@ -7,6 +7,27 @@ metadata:
   labels:
 {{ include "zookeeper.commonLabels" . | indent 4 }}
 spec:
+  initContainers:
```

Comment:
I think the logic that is present in the command of this init container is quite important, and should not be left to the submitter of the Zookeekeeper Cluster CR.

That init container logic is something that should be provided to anybody who submits a ZookeeperCluster for provisioning.  

If you look here: https://github.com/bitnami/charts/blob/master/bitnami/postgresql/templates/statefulset.yaml#L110, that initcontainer with similar logic is being embedded right into the statefulset.

In the postgresql case, it's part of the chart itself because the postgres instance is not a CR.   In here, I think it should be part of the zookeeper operator code that creates this statefulset  (generators.go, makeZkPodSpec) for the Zkcluster.    Meaning the chown logic etc. shouldn't be up to the end user.

There are additional nuances to this with OPenshift:   see how you have runAsUser: 0 in the securityContext of the init container ?    This could cause problems in some openshift environment if the serviceaccount for the pod (which applies to all containers of the pod including the init containers) has not been given access to the anyuid SCC.   (it is just not a guarantee at all).

I think a good path forward would be to imitate what postgresql did:

1. first, there have a boolean in to indicate where or not you want this volume permission initcontainer to be injected in the pod or not.
2. second, if you do want to have the initcontainer that sets permission, be aware that this might be in an openshift env., and so you might want to let Openshift decide what the "RunUserAs" is.   This comment explains it well https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml#L71
(in their case, they use this "auto" keyword to indicate to leave the RunAsUser blank when they copy the security context object)

So either have a RunUserAs SC that gets applied to both the initcontainer and the main container.   Or a securityContext with NO RunAsUser, in both, to let Openshift provide you with a uid, that you then just chown things to.

And again, because that logic is tricky, best to have the zk operator provide this when it's building the statefulset than leave it open for the submitted of the zkcluster CR who will for sure get it wrong. :)

Reply:
I have changed the approach as you suggested. Please take a look